discord-ptb: update to 0.0.22

MGlolenstine opened the pull request.

fosslinux commented on srcpkgs/discord-ptb/template:
```diff
@@ -22,9 +22,9 @@ do_install() {
 	vcopy discord-ptb.desktop /usr/share/applications/
 	vmkdir ${package_location}
-	for item in locales resources DiscordPTB libffmpeg.so libnode.so natives_blob.bin snapshot_blob.bin content_shell.pak icudtl.dat; do
+	for item in DiscordPTB chrome_100_percent.pak chrome_200_percent.pak icudtl.dat libEGL.so libGLESv2.so libffmpeg.so locales natives_blob.bin resources resources.pak snapshot_blob.bin swiftshader v8_context_snapshot.bin; do
```
Preferably, split this into two lines using a `\` at the end of the line to make it more readable.

MGlolenstine replied:
Sure! I'm unsure on the formatting though. How long should the line be? Usually those are split on 90 characters.

fosslinux commented:
Thanks for your PR!

https://travis-ci.org/github/void-linux/void-packages/jobs/715664522 as you can see here, there are some `xlint` erorrs. Would you mind fixing them?

MGlolenstine commented:
On it!

fosslinux commented:
The `custom:` speciefier indicates it is not SPDX. Use something along the lines of `custom:proprietary`.

The line wrap looks good, but maybe make it consistent. I.e., if you are going to split at (I think you did) 90 characters, split the next line at 90 characters too... keep all of the lines for that block the same length.

ericonr commented:
I believe "custom:Proprietary" is preferred. And line wraps around 80 columns, if possible.

MGlolenstine commented:
I think "custom:Proprietary" had some problems with xlint and was throwing errors... I'll have to check out that tomorrow (it's midnight here) and I'm in for 80 char line wrap.

MGlolenstine commented:
@ericonr This is the lint error I was talking about. I'm unsure what to put at the `vlicense` if anything.

ericonr commented:
You might need to find an EULA or License Agreement. The spotify template, for example, has to do a similar dance.

MGlolenstine commented:
> You might need to find an EULA or License Agreement. The spotify template, for example, has to do a similar dance.

Would this do? 
How should I format it in the file? 
```
vlicense=https://discord.com/terms
```

ericonr commented:
I think that file would work, but you'd have to do something like https://github.com/void-linux/void-packages/blob/master/srcpkgs/spotify/template#L23 to pre-process it.

MGlolenstine commented:
> I think that file would work, but you'd have to do something like https://github.com/void-linux/void-packages/blob/master/srcpkgs/spotify/template#L23 to pre-process it.

Oh wow, that looks complicated... It's pretty late here, so I'll take my time tomorrow and modify it...

MGlolenstine commented:
Would this be ready for merge, or is there something still missing?

FirstZero commented:
Ideally you might want to squash rebase your commits into one. Other than that, this seems ready for review.

ericonr commented:
You need to squash these commits before it can be merged.

MGlolenstine commented:
Ugh... I guess that squash didn't work like I wanted it to. :sweat_smile:

MGlolenstine commented:
This seems better.

Chocimier commented on srcpkgs/discord-ptb/template:
```diff
@@ -1,19 +1,31 @@
 distfiles="https://dl-ptb.discordapp.net/apps/linux/${version}/discord-ptb-${version}.tar.gz"
-checksum=372d6f75203626370218c97c21519b779e5af334177cafa65a5655e357c77520
+checksum=a1fa4e20cce5f493d84c9e3c7ac74c8265768886859e98022fa41d102f2726c4
+_license_checksum=e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
```
This is checksum of empty string. Empty file is produced because of
```
=> discord-ptb-0.0.21_1: running post_extract ...
/void-packages/srcpkgs/discord-ptb/template: line 21: w3m: command not found
/void-packages/srcpkgs/discord-ptb/template: line 20: curl: command not found
```

Add w3m to hostmakedepends, use `$XBPS_FETCH_CMD -o eula` instead of `curl -L`.

MGlolenstine replied:
I've installed both, so the errors weren't being thrown for me. Thanks for checking it out!

Chocimier replied:
license in package is still empty

MGlolenstine replied:
@Chocimier If you have any ideas on how to fix it, I'm welcoming any. I'm not very knowledgable and I don't know much about VoidLinux, but this above code works for me.

Please help me fix this template so we can merge this.

Chocimier replied:
`$XBPS_FETCH_CMD -o eula` outputs to eula file, not stdout. Do something like `$XBPS_FETCH_CMD -o eula; cat eula | ...`, rebuild (with local revbump possibly to create new package, but then send with revision=1) and check if packaged license has right content.

MGlolenstine replied:
I might be thick-skulled, but I'm not exactly sure why we don't want to save it to a file, but instead output it to a stdout. Is there a reason for it?

I'll try doing the above as soon as my laptop is free(few hours).

MGlolenstine replied:
I figured out that I was being thick-skulled and I realised what the problem with the file was...
Piping didn't do anything if the output of the first command wasn't outputted to the stdout.

In the meantime, there has been an update to 0.0.22, so I'll update to that and commit again.

Chocimier commented on srcpkgs/discord-ptb/template:
```diff
@@ -1,19 +1,31 @@
+post_extract() {
+	curl -L https://discord.com/terms |
+		w3m -dump -I utf-8 -T text/html |
+		sed -n '/Discord Terms of Service/,/^USA$/p' > EULA
```
There is no word "USA" in this license.

MGlolenstine replied:
You're right. I have no idea why I didn't replace that with something that's inside of the EULA.
Thanks for pointing it out!

chilledfrogs commented:
BTW, just a little detail, but they moved to https://discord.com/ recently-ish @MGlolenstine, might wanna update that in homepage (for the moment the old URL is still usable though so not a huge deal)

MGlolenstine commented:
> BTW, just a little detail, but they moved to https://discord.com/ recently-ish @MGlolenstine, might wanna update that in homepage (for the moment the old URL is still usable though so not a huge deal)

I've just updated the URL.

MGlolenstine commented:
I've updated the package to 0.0.22.
Should I close this PR and create a new one, as the version changed or would it be ok, if I just changed the commit message and PR's title?

fosslinux commented:
No, you should just change the commit message and the title.

MGlolenstine commented:
Should commit message contain every change or just the "discord-ptb: update to 0.0.22"?

fosslinux commented:
How you have it now is correct.

Chocimier commented:
@MGlolenstine Do you run this updated 0.0.22 version? Is everything ok?

MGlolenstine commented:
@Chocimier Correct, I've updated it because I needed it for my laptop. I'm running it and it's working perfectly fine.
